events/loginEvent.py

```diff
@@ -120,6 +120,19 @@ async def handle(web_handler: AsyncRequestHandler) -> tuple[str, bytes]:  # toke
             )
             raise exceptions.haxException()
 
+        # Only allow a single non-tourney session at a time.
+        # (Perhaps we could support some faster anti-ghosting here,
+        #  although the inactive-session-timeout cronjob handles it)
+        is_tournament_client = rgx["stream"] == "tourney"
+        if not is_tournament_client:
+            existing_primary_token = await osuToken.get_primary_token_by_user_id(userID)
+            if existing_primary_token:
+                logger.warning(
+                    "A user attempted to login with multiple primary sessions",
+                    extra={"user_id": userID, "username": username},
+                )
+                raise exceptions.primarySessionAlreadyExistsException()
```

Comment:
Behaviour change: instead of kicking all active tokens when logging in from a primary session, instead just disallow new primary logins, and let primary & non-primary sessions login in whichever order; they're effectively completely isolated